Devel::CheckLib is never successful for libssl on OpenBSD.  You
must specify -lssl and -lcrypto together, but Devel::CheckLib is
checking them individually.  So the dependency on libssl was missing
if p5-Crypt-SSLeay was compiled when p5-Devel-CheckLib was installed.

Use the libraries we have on OpenBSD and do not use Devel::CheckLib.

Index: Makefile.PL
--- Makefile.PL.orig
+++ Makefile.PL
@@ -8,10 +8,11 @@ use Getopt::Long qw( GetOptionsFromArray );
 use Path::Class;
 use Try::Tiny;
 
+BEGIN { push @INC, '.'; }
 use inc::IO::Interactive::Tiny;
 
 caller
-    or run(\@ARGV, [qw{ssl crypto ssl32 ssleay32 eay32 libeay32 z}]);
+    or run(\@ARGV, [qw{ssl crypto z}]);
 
 sub run {
     my $argv = shift;
@@ -75,8 +76,6 @@ sub run {
             exit 0;
         };
     }
-
-    $libs = filter_libs(\%opt, $libs);
 
     unless ($ENV{AUTOMATED_TESTING}) {
         show_do_you_need_crypt_ssleay();
